src/main/java/health/ere/ps/service/idp/BearerTokenService.java

```diff
@@ -39,13 +41,27 @@ public class BearerTokenService {
 
 
     @PostConstruct
-    public void init() {
+    public void init() throws Exception {
+        // Make this process sync to prevent Application start in dysfunctional state
+
+        List<Integer> retrySeconds = appConfig.getIdpInitializationRetriesSeconds();
+        int retryPeriodMs = appConfig.getIdpInitializationPeriodMs();
+        boolean initialized = Retrier.callAndRetry(retrySeconds, retryPeriodMs, this::initializeIdp, bool -> bool);
```

Comment:
This will block the whole start thread for a long time if it does not work. The retrier should use an own thread.

Reply:
Fixed